Zero/new chat input

zeroliu opened the pull request.

zeroliu commented:
@logancyang I realize the pdf context support is broken. Let me fix it

zeroliu commented:
@logancyang it's good now

logancyang commented:
@zeroliu This is a HUGE upgrade to our chat input, thanks for the hard work!! 🔥🔥🔥 

I clicked around a bit and here are some initial findings:

1. Seems I can't add a url as plain text. When i paste, it becomes a badge; when i backspace, it gets removed completely as a badge.
<img width="273" height="88" alt="SCR-20250924-okjv" src="https://github.com/user-attachments/assets/d8d23bfb-af93-487f-a357-c5a74fbaf41f" />

2. Typing `@` can enter the tool in the input, but clicking by mouse doesn't work.
<img width="591" height="359" alt="SCR-20250924-okcj" src="https://github.com/user-attachments/assets/60083806-a444-46e3-a845-dddfecb5c6bd" />

3. Adding notes or urls makes the "@ Add Context" button collapse to just "@", but Folders or Tags don't

<img width="348" height="86" alt="SCR-20250924-ojkx" src="https://github.com/user-attachments/assets/8d51e5b4-d644-4859-bd64-d9036b4ca345" />
<img width="332" height="84" alt="SCR-20250924-ojrc" src="https://github.com/user-attachments/assets/53e13ecb-0c00-4bc6-9053-2579b1428e38" />

4. When i type `[[` the popup shows 4-5 candidates with previews, this search works differently from the `@ Add Context` button. For example, if i type `[[pdf` it won't find pdf files, but `@ Add Context` can find them.

5. If i open the settings UI while having the context picker and preview on, they stay there like this
<img width="2595" height="1306" alt="SCR-20250924-opic" src="https://github.com/user-attachments/assets/4fb6a1a1-fe3a-498a-8f5c-7f80be2f705a" />

zeroliu commented:
> @zeroliu This is a HUGE upgrade to our chat input, thanks for the hard work!! 🔥🔥🔥
> 
> I clicked around a bit and here are some initial findings:
> 
> 1. Seems I can't add a url as plain text. When i paste, it becomes a badge; when i backspace, it gets removed completely as a badge.
> 
> <img alt="SCR-20250924-okjv" width="273" height="88" src="https://private-user-images.githubusercontent.com/4860545/493616331-d8d23bfb-af93-487f-a357-c5a74fbaf41f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg3ODI4OTYsIm5iZiI6MTc1ODc4MjU5NiwicGF0aCI6Ii80ODYwNTQ1LzQ5MzYxNjMzMS1kOGQyM2JmYi1hZjkzLTQ4N2YtYTM1Ny1jNWE3NGZiYWY0MWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MjVUMDY0MzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzBiNDMwMDZjYjc0MzBjYjk5MTgxYTI1MTJmNTY4YmU2OTczMWQ1ZGY4Y2IwNjNjYzcwYWY5ZmI1YmZlNDAyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gA6xp-M_4gK2Mo7LcLAUGKL1hc2Xrg5_r7Zcc75dq1E">

This one is intentional. I checked the latest cursor and it doesn't contain the old behavior of converting a URL pill to plain text any more. I suspect that if a URL is removed from the context, most likely you want to remove it from the context too because the URL itself doesn't add much value to the context. If we decide in the future to process the chat content and fetch the plain text URL, the current behavior is future-proof, too. WDYT?


> 2. Typing `@` can enter the tool in the input, but clicking by mouse doesn't work.
> 
> <img alt="SCR-20250924-okcj" width="591" height="359" src="https://private-user-images.githubusercontent.com/4860545/493616390-60083806-a444-46e3-a845-dddfecb5c6bd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg3ODI4OTYsIm5iZiI6MTc1ODc4MjU5NiwicGF0aCI6Ii80ODYwNTQ1LzQ5MzYxNjM5MC02MDA4MzgwNi1hNDQ0LTQ2ZTMtYTg0NS1kZGRmZWNiNWM2YmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MjVUMDY0MzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTIwNjIyMzJjOWM1NWFjNmIyMDI5ODExZWI5NmRkNGNkZGM2OWVmNjc2NmMzZDU3ZjNlMTc4MGY4ZTcwMjA4ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.bBqsjZOCrbGwnYe24sN33ZydHw30MBSK01ctsZ2AVsM">

It's actually working. The tool is toggled and the tool button is updated. This is intentional because I don't want the tool pill to show up in the context menu since the bottom tool button already shows that. I understand why it's misleading though. In the latest version, when you click on a tool option in the context menu, it will add a tool pill in the chat. This would work for @memory where there is no tool button. It's still a bit different from other type of context but I think it's the best option so far.

> 3. Adding notes or urls makes the "@ Add Context" button collapse to just "@", but Folders or Tags don't
> 
> <img alt="SCR-20250924-ojkx" width="348" height="86" src="https://private-user-images.githubusercontent.com/4860545/493616826-8d51e5b4-d644-4859-bd64-d9036b4ca345.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg3ODI4OTYsIm5iZiI6MTc1ODc4MjU5NiwicGF0aCI6Ii80ODYwNTQ1LzQ5MzYxNjgyNi04ZDUxZTViNC1kNjQ0LTQ4NTktYmQ2NC1kOTAzNmI0Y2EzNDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MjVUMDY0MzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFhZmRlNWM2ZTI1NDMwN2NlNTcwNGViYzY1Y2Q4OTNhYzU0YzRiNWI3YmEwNjRhOTA5YTVjY2Q0MDU1NzNmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.XgGl6mDNU3h9hUVMY0OU-WmT25zS3Wr2rke5TKpokpA"> <img alt="SCR-20250924-ojrc" width="332" height="84" src="https://private-user-images.githubusercontent.com/4860545/493616894-53e13ecb-0c00-4bc6-9053-2579b1428e38.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg3ODI4OTYsIm5iZiI6MTc1ODc4MjU5NiwicGF0aCI6Ii80ODYwNTQ1LzQ5MzYxNjg5NC01M2UxM2VjYi0wYzAwLTRiYzYtOTA1My0yNTc5YjE0MjhlMzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MjVUMDY0MzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDg5YTUwNjcwYzI1OTUzNmU2MjEzYzkwYWRiYWJkYmZhYjBmZDllMDFiM2IyYjY4N2E1ODhlN2QyM2FmZGUxOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4i8nUukGBNM7xwxh1bQsGF4m7-flqRj05iJSrPn-5Y0">

Fixed!

> 4. When i type `[[` the popup shows 4-5 candidates with previews, this search works differently from the `@ Add Context` button. For example, if i type `[[pdf` it won't find pdf files, but `@ Add Context` can find them.
> 5. If i open the settings UI while having the context picker and preview on, they stay there like this
> 
> <img alt="SCR-20250924-opic" width="2000" height="1306" src="https://private-user-images.githubusercontent.com/4860545/493619991-4fb6a1a1-fe3a-498a-8f5c-7f80be2f705a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg3ODI4OTYsIm5iZiI6MTc1ODc4MjU5NiwicGF0aCI6Ii80ODYwNTQ1LzQ5MzYxOTk5MS00ZmI2YTFhMS1mZTNhLTQ5OGEtOGY1Yy03ZjgwYmUyZjcwNWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MjVUMDY0MzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDg5MDIzZjk4NmYxNTBmYTc2MjAzMDlkYzA3ZDM0MzY0YTE2MDJhZmNhMjAyMjMxZTRkODc1ZmJkZGYxMzU2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.MVXTitO-zrKDpblaOgHt6uUBdXRuJqdlHms772iuIKE">

Fixed!

> 6. Could you remind me what we have settled on for the old `@youtube <url>` behavior - getting full raw transcript? How do we achieve it now?

We decided to introduce a new download youtube video script command. 

<img width="1060" height="844" alt="Screenshot 2025-09-25 at 1 01 30 AM" src="https://github.com/user-attachments/assets/ee1e25cd-2176-4780-ae49-0975b7b6a851" />

<img width="634" height="623" alt="Screenshot 2025-09-25 at 1 01 49 AM" src="https://github.com/user-attachments/assets/4afab843-790b-4dbb-8539-b12ae9104c21" />

Emt-lin commented:
1. This should be a display bug. When I hover over an item and then use the arrow keys to move up or down, the hover state of the item under the mouse remains.
<img width="500" height="500" alt="Obsidian 2025-09-26 23 37 42" src="https://github.com/user-attachments/assets/e6c2dd91-7b68-42ea-b64d-0202b888fbd6" />

2. on mobile,  there are some UI  issues for the `@` portal.

<img width="914" height="844" alt="CleanShot 2025-09-27 at 00 02 49" src="https://github.com/user-attachments/assets/28a108f1-bc36-47b3-8dbb-681717866f69" />

3. for the `@` context portal, maybe It is a little wide. Is it to previe the note content?

<img width="600" height="800" alt="CleanShot 2025-09-27 at 00 10 37" src="https://github.com/user-attachments/assets/f9b83f88-4361-495d-a86e-ec2235c9319f" />
